packages/storybook/src/executors/storybook/storybook.impl.ts

```diff
@@ -42,6 +42,25 @@ export default async function* storybookExecutor(
 function runInstance(options: StorybookExecutorOptions) {
   const env = process.env.NODE_ENV ?? 'development';
   process.env.NODE_ENV = env;
+  const originalConsoleLog = console.log;
+  console.log = function (...args) {
+    const localhostString = 'http://localhost:';
+    if (args[0].includes(localhostString)) {
+      const indexOfLocalHost = args[0].indexOf(localhostString);
+      console.info(
+        args[0].slice(
+          indexOfLocalHost + localhostString.length,
+          indexOfLocalHost + localhostString.length + 4
+        ),
+        indexOfLocalHost
+      );
+      process.env.NXPORT = args[0].slice(
+        indexOfLocalHost + localhostString.length,
+        indexOfLocalHost + localhostString.length + 4
+      );
+    }
```

Comment:
I'll change this to a regex or something similar to make sure just to take the numbers after the localhost